pyanaconda/ui/gui/spokes/network.py

```diff
@@ -1289,47 +1285,61 @@ def _update_connect_button(self):
 secret_agent = None
 
 
-class NotAuthorizedException(dbus.DBusException):
-    _dbus_error_name = SECRET_AGENT_IFACE + '.NotAuthorized'
-
+class SecretAgent(object):
+    __dbus_xml__ = """
+    <node>
+        <interface name="org.freedesktop.NetworkManager.SecretAgent">
+            <method name="GetSecrets">
+                <arg name="connection" type="a{sa{sv}}" direction="in"/>
+                <arg name="connection_path" type="o" direction="in"/>
+                <arg name="setting_name" type="s" direction="in"/>
+                <arg name="hints" type="as" direction="in"/>
+                <arg name="flags" type="u" direction="in"/>
+                <arg name="secrets" type="a{sa{sv}}" direction="out"/>
+            </method>
+            <method name="SaveSecrets">
+                <arg name="connection" type="a{sa{sv}}" direction="in"/>
+                <arg name="connection_path" type="o" direction="in"/>
+            </method>
+        </interface>
+    </node>
+    """
 
-class SecretAgent(dbus.service.Object):
     def __init__(self, spoke):
-        self._bus = dbus.SystemBus()
         self.spoke = spoke
-        super().__init__(self._bus, "/org/freedesktop/NetworkManager/SecretAgent")
-
-    @dbus.service.method(SECRET_AGENT_IFACE,
-                         in_signature='a{sa{sv}}osasb',
-                         out_signature='a{sa{sv}}',
-                         sender_keyword='sender')
-    def GetSecrets(self, connection_hash, connection_path, setting_name, hints, flags, sender=None):
-        if not sender:
-            raise NotAuthorizedException("Internal error: couldn't get sender")
-        uid = self._bus.get_unix_user(sender)
-        if uid != 0:
-            raise NotAuthorizedException("UID %d not authorized" % uid)
-
-        log.debug("secrets requested path '%s' setting '%s' hints '%s' new %d",
+
+    def GetSecrets(self, connection_hash, connection_path, setting_name, hints, flags):
+        log.debug("GetSecrets: secrets requested path '%s' setting '%s' hints '%s' new %d",
                   connection_path, setting_name, str(hints), flags)
         if not (flags & NM_SECRET_AGENT_GET_SECRETS_FLAG_ALLOW_INTERACTION):
             return
 
-        content = self._get_content(setting_name, connection_hash)
+        unpacked_connection_hash = self._unpack_connection_hash(connection_hash)
+
+        content = self._get_content(setting_name, unpacked_connection_hash)
         dialog = SecretAgentDialog(self.spoke.data, content=content)
         with self.spoke.main_window.enlightbox(dialog.window):
```

Comment:
If you replace the spoke with a callback, you might be able to write unit tests for this class.